sys/riscv/riscv/locore.S

```diff
@@ -108,7 +108,7 @@ _start:
 	add	t3, t4, t2
 	li	t5, 0
 2:
-	li	t0, (PTE_KERN | PTE_X)
+	li	t0, (PTE_KERN | PTE_X | PTE_LC | PTE_SC)
```

Comment:
Jenkins gives:
```
.../locore.S:111:9: error: operand must be a constant 64-bit integer
 li t0, (((1 << 0) | (1 << 1) | (1 << 2) | (1 << 6) | (1 << 7)) | (1 << 3) | PTE_LC | PTE_SC)
        ^
```

There needs to be an #ifdef involved here; you don't get default 0 expansion like you do in `#if`.

Reply:
Ok.  I kind of like the bits being explicit, but I think it's also probably better to have PTE_LC and PTE_SC not exist for non-CHERI.  I don't think we need both PTE_KERNCAP and PTE_KERNNOCAP as PTE_KERN can be one of those.  I think I probably prefer to add a PTE_KERNCAP (or PTE_KERN_CAP) that includes LC/SC for CHERI and is just PTE_KERN for !CHERI.

Reply:
Actually, this is the only place we'd have to #ifdef since the rest are in C, so perhaps I can just #ifdef here?

Reply:
pmap_bootstrap_dmap and pmap_qenter will need #ifdef's too unless you add PTE_KERN_CAP. Otherwise they will error with an unknown identifier. It's only `#if` etc that implicitly turn undefined things into 0; everything else in the preprocessor has to leave them untouched otherwise every identifier would become 0.